src/ui.py

```diff
@@ -278,3 +279,14 @@ def bind_values_to_object(source, window, ignorelist=None):
         except AttributeError:
             logging.warn(_("No way to apply “{}”").format(key))
     return bindings
+
+
+def reload_canvas(self):
```

Comment:
The reason was, to move Canvas instantiation out of the window class to enable porting it to Vala

Reply:
> I was planning to inherit that canvas from the main canvas class, and suspect a reload_canvas function would be useful there. Any specific reason to move this to main?

I have moved the property bindings to the ui function instead, this way the main canvas class could be used more freely. However reloading the canvas can only be implemented in a super function (previously the main window). For A Curve Fitting window, there needs to be a specific implementation for that. Given the move of the bindings we can also restrict that to just show for example the item and the suggested curve for easier visual comparison.

Long story short. This aims to make the canvas Widget a bit more versatile so it could be used idaeally without modification.